compiler/damlc/daml-package-config/src/DA/Daml/Package/Config.hs

```diff
@@ -66,6 +68,25 @@ parseProjectConfig project = do
     pSdkVersion <- queryProjectConfigRequired ["sdk-version"] project
     Right PackageConfigFields {..}
 
+checkPkgConfig :: PackageConfigFields -> [String]
+checkPkgConfig PackageConfigFields {pName, pVersion} =
+  [ "WARNING: Package names should have the format ^[a-zA-Z]([a-zA-Z0-9\\-])*[a-zA-Z0-9]$. \
+  \ You may be able to compile packages with different formats, but you will not be able to use \
+  \ them as dependencies in other projects. Unsupported package names may start causing \
+  \ compilation errors without warning."
+  | not $ LF.unPackageName pName =~ ("^[a-zA-Z]([a-zA-Z\\-])*[a-zA-Z]$" :: T.Text)
```

Comment:
This still has the issue with two consecutive dashed. I'd suggest we go for
```suggestion
  | not $ LF.unPackageName pName =~ ("^[A-Za-z][A-Za-z0-9]*(\-[A-Za-z][A-Za-z0-9]*)*$" :: T.Text)
```
I'd also prefer to bind the regular expression to a variable (perhaps of type `IsString s => s`, if necessary) and use that variable in the message and on the RHS of `=~` to avoid that they diverge. (In fact, they are already different right now.) Let's do this below as well.

Reply:
I already changed the regex, so not commiting this suggestion. I'll bind the regex to a variable.